structure.go

```diff
@@ -114,3 +118,81 @@ func (s Structure) UsePalette(name string) {
 	s.paletteName = name
 	s.parsePalette()
 }
+
+// RotateLeft returns a new structure with the same contents but rotated 90 degrees anti-clockwise.
+func (s Structure) RotateLeft() Structure {
+	return s.rotate(-1)
+}
+
+// RotateRight returns a new structure with the same contents but rotated 90 degrees clockwise.
+func (s Structure) RotateRight() Structure {
+	return s.rotate(1)
+}
+
+// rotate returns a new structure with the same contents but rotated 90 degrees in the specificed direction.
+func (s Structure) rotate(direction int) Structure {
+	sizeX, sizeY, sizeZ := int(s.Size[0]), int(s.Size[1]), int(s.Size[2])
+	newStructure := New([3]int{sizeZ, sizeY, sizeX})
+
+	maxX, maxZ := sizeX-1, sizeZ-1
+	for x := 0; x < sizeX; x++ {
+		for y := 0; y < sizeY; y++ {
+			for z := 0; z < sizeZ; z++ {
+				newX, newZ := x, z
+				if direction == 1 {
+					newX = -z + maxZ
+					newZ = x
+				} else {
+					newX = z
+					newZ = -x + maxX
+				}
+				b, l := s.At(x, y, z, nil)
+				newStructure.Set(newX, y, newZ, b, l)
+			}
+		}
+	}
+	for i, state := range s.palette.BlockPalette {
+		b, ok := world.BlockByName(state.Name, state.States)
+		if !ok {
+			continue
+		}
+
+		origin := reflect.ValueOf(b)
+		t := reflect.TypeOf(b)
+		v := reflect.New(t).Elem()
+
+		for i := 0; i < v.NumField(); i++ {
+			fieldV := v.Field(i)
+			if !ast.IsExported(t.Field(i).Name) {
+				continue
+			}
+
+			// We set the field of the original to the new one to make sure that the existing values are placed
+			// back in the form element. Not doing so would result in unexpected behaviour.
+			fieldV.Set(origin.Field(i))
+
+			switch t := fieldV.Interface().(type) {
```

Comment:
I'd also add support for `cube.Face` and `cube.Orientation`. Both of those types also have a `RotateRight()` and `RotateLeft()` method.

Reply:
You could consider using `(reflect.Value).MethodByName(...).Call()` to call the `RotateRight()` and `RotateLeft()` methods for all of these types in the same way. All of them have these methods.